src/main/java/com/auth0/AuthorizeUrl.java

```diff
@@ -13,18 +17,26 @@
 public class AuthorizeUrl {
 
     private static final String SCOPE_OPENID = "openid";
-    private final HttpServletRequest request;
+    private final HttpServletResponse response;
     private final AuthorizeUrlBuilder builder;
+    private final String responseType;
+    private boolean legacySameSiteCookie;
+    private String nonce;
+    private String state;
+
     private boolean used;
 
     /**
      * @param client       the Auth0 Authentication API client
-     * @param request      request where the state will be saved
+     * @param request      the Http request
+     * @param response     the response where the state and nonce will be stored as cookies
      * @param redirectUrl  the url to redirect to after authentication
      * @param responseType the response type to use
      */
-    AuthorizeUrl(AuthAPI client, HttpServletRequest request, String redirectUrl, String responseType) {
-        this.request = request;
+    AuthorizeUrl(AuthAPI client, HttpServletRequest request, HttpServletResponse response, String redirectUrl, String responseType) {
```

Comment:
Maintaining `HttpServletRequest` parameter for potential future use without introducing another potentially breaking change.